### PR TITLE
Edit 3rd book

### PR DESCRIPTION
**Pull request by RichardKenner**

### CODE REVIEW

File: courses/intro-to-spark/book/03_Proof_Of_Program_Integrity.rst [hunk 1]
```diff
@@ -347,36 +355,37 @@ below:
...
 
-New contracts have also been introduced for the process of formal
-verification. For example, the new pragma :ada:`Assume` is handled as an
-assertion at execution but introduces an assumption for proof of program,
-that is, a Boolean expression which is assumed to be true by GNATprove
-without any verification. This feature is useful but must be used with
-great care.
+The SPARK language also includes new contract types used to assist formal
+verification. The new pragma :ada:`Assume` is treated as an assertion
+during execution but introduces an assumption when proving programs.  It's
```
Comment (gusthoff) [inline courses/intro-to-spark/book/03_Proof_Of_Program_Integrity.rst]: *Its

File: courses/intro-to-spark/book/03_Proof_Of_Program_Integrity.rst [hunk 2]
```diff
@@ -386,13 +395,15 @@ great care.
...
-aspect. It allows to specify the behavior of a subprogram with a disjunction of
-cases. Each element of a contract-cases is in fact a small contract made of a
-guard, which may only reference subprogram's inputs and is evaluated before the
-call, and of a consequence. At each call of the subprogram, there must be one
-and only one case for which the guard evaluates to :ada:`True`. The consequence
-of this case is the one that should hold on exit.
+The :ada:`Contract_Cases` aspect is another construct introduced for
+GNATprove, but which also acts as an assertion during execution. It allows
+you to specify the behavior of a subprogram using a disjunction of
+cases. Each element of a :ada:`Contract-Cases` aspect is a *guard*, which
```
Comment (gusthoff) [inline courses/intro-to-spark/book/03_Proof_Of_Program_Integrity.rst]: *_Contract_Cases_

File: courses/intro-to-spark/book/03_Proof_Of_Program_Integrity.rst [hunk 3]
```diff
@@ -466,11 +480,11 @@ there is an error in our procedure ``Incr_Until`` which makes its
...
     end Show_Failed_Proof_Attempt;
 
-As assertions can be executed, it may help to test the program on a
-representative set of inputs with assertions enabled. This allows bugs to
-be found both in the code and in its contracts. For example, testing
-``Incr_Until`` on an input bigger than 1000 will raise an exception at
-runtime.
+Since this is an assertion that can be executed, it may help you find the
+problem if you run the program with assertions enabled on representative
+sets of inputs . This allows you to find bugs in both the code and its
```
Comment (gusthoff) [inline courses/intro-to-spark/book/03_Proof_Of_Program_Integrity.rst]: Minor editorial remark: there's an extraneous whitespace after *sets of inputs*

**Comment by RichardKenner** [comment]
> *can ever be

Got it.

> *Its

Got it.

> *Contract_Cases

That too.

>   +sets of inputs . This allows you to find bugs in both the code and its
> 
> Minor editorial remark: there's an extraneous whitespace after sets of inputs

And this one.  And I checked to see if there were any more (there aren't).

I updated the branch.

**Pull request merged**